### ISSUE TITLE
Loose coupling to contrib.sites application

### ISSUE THREAD

**Issue by floledermann** [flashingpumpkin/django-socialregistration#90]
django-socialregistration is currently hard-wired to the sites application. This means it cannot be used in scenarios where the sites app is not installed, or where other methods for assigning authenticated users to sites are used.

**Comment by flashingpumpkin** [comment]
This is true. I've made it that way because the auth methods require to know where to redirect back. 

A solution would be to turn it into some kind of internal `Site` model, that has a domain field which in turn can take a default value from a function, which can be defined in the settings - similar to upload paths for files. And if not, it's using `Site.objects.get_current().domain`
